src/Mod/PartDesign/Gui/TaskBooleanParameters.cpp

```diff
@@ -166,10 +166,9 @@ void TaskBooleanParameters::onSelectionChanged(const Gui::SelectionChanges& msg)
 
                 QString internalName = QString::fromStdString(body);
                 for (int row = 0; row < ui->listWidgetBodies->count(); row++) {
-                    QListWidgetItem* item = ui->listWidgetBodies->item(row);
+                    QListWidgetItem* item = ui->listWidgetBodies->takeItem(row);

```

Comment:
I suspect the replacement of this `item` call with `takeItem` is incorrect.
A major difference between the change to use `takeItem` in 2b67602 vs. this usage is that in the referenced commit the `item` is always deleted if a non-`nullptr`. In this case it is should only be removed if `name == internalName`.

Reply:
Yes, using takeItem() in this case is incorrect. If the user has selected the last item in the list this change effectively empties it and even worse causes a serious memory leak because when taking an item from a list it's not managed by it any more.
This is clearly described in the Qt docs: https://doc.qt.io/qt-5/qlistwidget.html#takeItem